internal/clearingway/handlers.go

```diff
@@ -466,6 +517,79 @@ func (c *Clearingway) RemoveAll(s *discordgo.Session, i *discordgo.InteractionCr
 	}
 }
 
+func (c *Clearingway) RequestReclear(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	g, ok := c.Guilds.Guilds[i.GuildID]
+	if !ok {
+		fmt.Printf("Interaction received from guild %s with no configuration!\n", i.GuildID)
+		return
+	}
+
+	// Ignore messages not on the correct channel
+	if i.ChannelID != g.ChannelId {
+		fmt.Printf("Ignoring message not in channel %s.\n", g.ChannelId)
+	}
+
+	err := discord.StartInteraction(s, i.Interaction, "Checking for respective clear role...")
+	if err != nil {
+		fmt.Printf("Error sending Discord message: %v\n", err)
+		return
+	}
+
+	ultimate := i.ApplicationCommandData().Options[0].StringValue()
+	encounter := g.Encounters.ForName(ultimate)
+	reclearRole := encounter.Roles[ReclearRole]
+	clearedRole := encounter.Roles[ClearedRole]
+
+	var rolePresent = false
+	var clearPresent = false
+	for _, r := range i.Member.Roles {
+		if r == reclearRole.DiscordRole.ID {
+			rolePresent = true
+			continue
+		}
+		if r == clearedRole.DiscordRole.ID {
+			clearPresent = true
+			continue
+		}
+	}
+
+	// Remove role no matter what
+	// Add role only if cleared role is present
+	if rolePresent {
+		fmt.Printf("Removing role: %+v\n", reclearRole.Name)
+		err = reclearRole.RemoveFromCharacter(g.Id, i.Member.User.ID, c.Discord.Session)
+		if err != nil {
+			fmt.Printf("Error removing role: %+v\n", err)
+			return
+		}
+		tempstr := fmt.Sprintf("Successfully removed role: <@&%v>", reclearRole.DiscordRole.ID)
+		err = discord.ContinueInteraction(s, i.Interaction, tempstr)
+		if err != nil {
+			fmt.Printf("Error sending Discord message: %v\n", err)
+		}
```

Comment:
this might bloat the role logging channel when we implement it. i think it would be better to only remove it if there is no clear role

Reply:
This is by design, people need a way to toggle the role on/off since it's gonna be pingable. The logic is set up such that if someone somehow has the reclear role without the clear role, they still have the option to toggle it off

Reply:
that shouldn't be a valid option though. the restraint is that they can only have the reclear role if they have the clear role

Reply:
It can be possible in very specific cases I think. If someone ran /clears on a different character for whatever reason and that character didn't have it, they would have the clear role removed. Plus I don't think it would bloat the role logging channel since that situation would be rare if it happens at all. Best case scenario is it spares us a ticket asking us to manually remove it and there's not really much downside pretty sure (maybe I'm missing something though?)

Reply:
i'm pretty sure clearingway only supports one character though, but it would bloat the role logging channel a lot because we will constantly have messages saying that it a role has been removed. how we have the code right now, we will always remove the role

Reply:
dont worry about the role bloat, we will already have it from the twitch live role

Reply:
this is just the past 2 hours
![image](https://github.com/user-attachments/assets/8a3ceaef-6cc9-43b0-b552-d0946588b8b9)

Reply:
Perhaps I'm misunderstanding, but it's only supposed to remove the role if the reclear role is present on you. I don't really see how that would bloat the logging channel

Reply:
also clearingway isnt being run enough to really be an issue we cant parse ourselves.
![image](https://github.com/user-attachments/assets/01fe0a01-66ab-4412-96eb-9351f8855133)
discord makes it very easy to see with built in audit log

Reply:
And to add another comment about logging, i do not think we will pursue what i envisioned before. i only care what Moddingway is doing. We can just use Carlbot since we need it anyways to make sure Exile is a sticky role.

Reply:
resolved in staff chat